ibis/backends/conftest.py

```diff
@@ -258,10 +443,464 @@ def pytest_runtest_call(item):
+                URLLIB_DEFAULT_POOL_SIZE,
+            )
+        )
+    )
+
+    hdfs = con.hdfs
+    tasks = {
+        # make the database
+        executor.submit(impala_create_test_database, con, env),
+        # build and upload UDFs
+        *itertools.starmap(
+            executor.submit,
+            impala_build_and_upload_udfs(hdfs, env, fs=fs),
+        ),
+        # upload data files
+        *(
+            executor.submit(
+                hdfs_make_dir_and_put_file,
+                hdfs,
+                data_file,
+                os.path.join(
+                    env.test_data_dir,
+                    os.path.relpath(data_file, data_dir),
+                ),
+            )
+            for data_file in data_files
+        ),
+    }
+
+    for future in concurrent.futures.as_completed(tasks):
+        future.result()
+
+    # create the tables and compute stats
+    for future in concurrent.futures.as_completed(
+        executor.submit(table_future.result().compute_stats)
+        for table_future in concurrent.futures.as_completed(
+            impala_create_tables(con, env, executor=executor)
+        )
+    ):
+        future.result()
+
+
+def load_pandas(data_dir, script_dir):
+    """No-op to allow pytest -s -m pandas"""
+
+
+def load_dask(data_dir, script_dir):
+    """No-op to allow pytest -s -m dask"""
+
+
+def load_datafusion(data_dir, script_dir):
+    """No-op to allow pytest -s -m datafusion"""
+
+
+def load_pyspark(data_dir, script_dir):
+    """No-op to allow pytest -s -m pyspark"""
+
+
+def load_duckdb(data_dir, script_dir):
+    """Load testdata into an impala backend.
+
+    Parameters
+    ----------
+    data_dir : Path
+        Location of testdata
+    script_dir : Path
+        Location of scripts defining schemas
+    """
+
+    IBIS_TEST_DUCKDB_DB = "ibis_testing"
+
+    schema = open(script_dir / 'schema' / 'duckdb.sql', 'r')
+    tables = TEST_TABLES
+
+    import duckdb  # noqa: F401
+
+    conn = duckdb.connect(str(data_dir / f"{IBIS_TEST_DUCKDB_DB}.ddb"))
+    for stmt in filter(None, map(str.strip, schema.read().split(';'))):
+        conn.execute(stmt)
+
+    for table in tables:
+        src = data_dir / f'{table}.csv'
+        conn.execute(
+            f"COPY {table} FROM '{src}' (DELIMITER ',', HEADER, SAMPLE_SIZE 1)"
+        )
+
+
+DB_LOADERS = {
+    "postgres": load_postgres,
+    "sqlite": load_sqlite,
+    "mysql": load_mysql,
+    "clickhouse": load_clickhouse,
+    "impala": load_impala,
+    "pandas": load_pandas,
+    "dask": load_dask,
+    "datafusion": load_datafusion,
+    "pyspark": load_pyspark,
+    "duckdb": load_duckdb,
+}
+
+
 @pytest.fixture(params=_get_backends_to_test(), scope='session')
-def backend(request, data_directory):
-    """Return an instance of BackendTest."""
+def backend(request, data_directory, script_directory):
+    """Return an instance of BackendTest, loaded with data."""
+    # TODO: Will request.param ever be a list of backends?
```

Comment:
TODO should be removed before merge: add an assert or handle the case for list of backends.